pixeltable/catalog/table.py

```diff
@@ -1550,3 +1552,131 @@ def __dir__(self) -> list[str]:
 
     def _ipython_key_completions_(self) -> list[str]:
         return list(self._schema.keys())
+
+    @classmethod
+    def diff_md(cls, old_md: Optional[dict[str, Any]], new_md: Optional[dict[str, Any]]) -> str:
+        """Return a string reporting the difference between two schema versions"""
+        assert new_md is not None
+        if old_md is None:
+            return 'Initial Version'
+        if old_md == new_md:
+            return ''
+        r = ''
+        added = {k: v['name'] for k, v in new_md.items() if k not in old_md}
+        if len(added) > 0:
+            r += 'Added: ' + ' '.join(added.values())
+        changed = {
+            k: old_md[k]['name'] + ' to ' + v['name']
+            for k, v in new_md.items()
+            if k in old_md and old_md[k]['name'] != v['name']
+        }
+        if len(changed) > 0:
+            r += 'Changed: ' + ' '.join(changed.values())
+        deleted = {k: v['name'] for k, v in old_md.items() if k not in new_md}
+        if len(deleted) > 0:
+            r += 'Deleted: ' + ' '.join(deleted.values())
+        return r
+
+    @classmethod
+    def create_md_change_dict(cls, src_rows: list[list[Any]]) -> dict[int, str]:
+        """Return a dictionary of schema changes by version given a list of version, column md"""
```

Comment:
Wait, does `src_rows` not contain table rows? The word `rows` is so hard-wired in pxt to mean "table rows" that I find that quite confusing.

Reply:
Ok, now I see how this is getting called; please please use something other than a `list[Any]` for the intermediate data structure. (A named tuple or similar struct is one option; if that seems too unwieldy, a typed tuple with appropriate code comments would still be way clearer than a list[Any].) That would make this code substantially easier to read.